### DISCUSSION TITLE
(Social) In-game chat extra options for hyperlinked content

### DISCUSSION THREAD

**Discussion by Dimmitsaras** [ppy/osu#23328] [Ideas] [answered]
Scenario: Your friend sends you a /np message to link a song they're playing/listening to.

Currently the only way to interact with the message content is
1) Hover over the song name to see the full URL
2) Left click to open the beatmap listing on the in-game browser

I think it would be cool to have a right click option, for example a context menu "Copy URL" or "Open in browser", this can help with scenarios like sharing the link to other friends/chatrooms, given that third party links already provide both of these options.

**Comment by Joehuu** [comment]
I agree links should be right-clickable for less navigation. A better scenario is for the wiki.

For your scenario, it is already possible to get the beatmap link by right-clicking this on the beatmap overlay:
![osu!_TcdfRaFnYQ](https://user-images.githubusercontent.com/35318437/235009876-330ba4b5-1f5e-4131-990d-5465becc8a70.png)

**Reply by Dimmitsaras** [reply]
Yeah thats pretty much what i was looking for but missed the option, I think my first instinct was to look for it here instead 
![image](https://user-images.githubusercontent.com/35996393/235071233-680e8c8b-429b-4250-b88f-0d993f843264.png)

Good to know!

**Reply by peppy** [reply, marked as answer]
I'm fine with right click support being added.

**Reply by Joehuu** [reply]
https://github.com/ppy/osu/pull/23334
